File: 3511-FindTheWinningPlayerInCoinGame/3511-FindTheWinningPlayerInCoinGame.java
// Last updated: 11/07/2026, 12:31:50
class Solution {
    public String winningPlayer(int x, int y) {
        int turns = (x < y / 4) ? x : (y / 4);
        return (turns % 2 == 1) ? "Alice" : "Bob";
    }
}

    

File: 3511-FindTheWinningPlayerInCoinGame/3511-FindTheWinningPlayerInCoinGame.cpp
// Last updated: 11/07/2026, 12:31:42
class Solution {
public:
    string winningPlayer(int x, int y) {
        int turn=1;
        while(x>=1 && y>=4){
            x-=1;
            y-=4;
            turn++;
        }
         if(turn%2==0) return "Alice";
         else return "Bob";
        
    }
};